lgc/interface/lgc/BuilderCommon.h

```diff
@@ -31,19 +31,20 @@
 #pragma once
 
 #include "llvm/IR/IRBuilder.h"
+#include "llvm-dialects/Dialect/Builder.h"
 
 namespace lgc {
 
 // =====================================================================================================================
-// BuilderCommon extends IRBuilder<>, and provides a few utility methods used in both the LLPC front-end and in LGC
-// (the LLPC middle-end).
+// BuilderCommon extends llvm_dialects::Builder, which extends llvm::IRBuilder<>, and provides a few utility methods
+// used in both the LLPC front-end and in LGC (the LLPC middle-end).
 // This class is used directly by passes in LGC.
-class BuilderCommon : public llvm::IRBuilder<> {
+class BuilderCommon : public llvm_dialects::Builder {
```

Comment:
(Further to comment above) So is the idea that the Builder interface itself will become a dialect? I guess that is not happening yet in these initial changes. Will that lose the current scheme that the front-end can choose to generate code directly at the Builder interface instead of using the record-replay mechanism? (The front-end currently never does that.)

Reply:
The idea is that the Builder can be used to create dialect ops directly, via `builder.create<FooOp>`, and that this capability is available in both LLPC and LGC.

Yes, this does mean de-emphasizing the record-replay mechanism, and while the two are *technically* orthogonal, I do think it's best to phase that mechanism out.

Reply:
Note that this is motivated by looking at where the LLVM ecosystem is going overall. With the `builder.create<FooOp>` pattern I'm really just copying what MLIR does. The implementation of the `create` template under the hood is different, and the details of how the create function arguments interact with the `.td` file definitions is also different (at least for now, it probably makes sense to align this better with MLIR), but the point is that the overall approach very much tries to be aligned with what others in the LLVM world are doing.